File: bkp/codewars/_bkp/004-remove-first-last-chars/typescript/solution1.ts
const l = console.log.bind(console);

/**
 * Removes first and last chars of the input string.
 *
 * ASSUME: Input string is never less than 3 chars long.
 */
export function chopSides(s: string): string {
  //
  // `slice` goes from `iniIndex` (inclusive) to `endIndex` (exclusive).
  // -1 means the index of the last char. Since `endIndex` is exclusive,
  // that index is not returned in the resulting string, effectively
  // excluding the last char.
  //
  // Also, `Array.prototype.slice` is pure; it _does not_ modify the
  // original array. It returns a copy with specified indexes.
  //
  return s.slice(1, -1);
};

l(chopSides('eloquent'));
// ⇒ loquen

l(chopSides('country'));
// ⇒ ountr

l(chopSides('person'));
// ⇒ erso

l(chopSides('place'));
// ⇒ lac

File: bkp/codewars/_bkp/004-remove-first-last-chars/ruby/solution1.rb

##
# Removes the first and last chars of a string.
#
# ASSUME: string is at least three chars long.
#
# @param [string] s
# @return [string]
#
def chop_sides(s)
  s[1...-1]
end

p chop_sides('eloquent')
# ⇒ loquen

p chop_sides('country')
# ⇒ ountr

p chop_sides('person')
# ⇒ erso

p chop_sides('place')
# ⇒ lac

